src/detect-tls-cert-serial.c

```diff
@@ -134,6 +144,74 @@ static InspectionBuffer *GetData(DetectEngineThreadCtx *det_ctx,
     return buffer;
 }
 
+static _Bool DetectTlsSerialValidateCallback(const Signature *s,
+                                             const char **sigerror)
+{
+    const SigMatch *sm = s->init_data->smlists[g_tls_cert_serial_buffer_id];
+    for ( ; sm != NULL; sm = sm->next)
+    {
+        if (sm->type != DETECT_CONTENT)
+            continue;
+
+        DetectContentData *cd = (DetectContentData *)sm->ctx;
+
+        if (cd->flags & DETECT_CONTENT_NOCASE) {
+            *sigerror = "tls_cert_serial should not be used together "
+                        "with nocase, since the rule is automatically "
+                        "uppercased anyway which makes nocase redundant.";
+            SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
+        }
+
+        /* no need to worry about this if the content is short enough */
+        if (cd->content_len <= 2)
+            return TRUE;
+
+        uint32_t u;
+        for (u = 0; u < cd->content_len; u++)
+            if (cd->content[u] == ':')
+                return TRUE;
+
+        *sigerror = "No colon delimiters ':' detected in content after "
+                    "tls_cert_serial. This rule will therefore never "
+                    "match.";
+        SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
```

Comment:
this is a warning but this leads to a rule loading error, correct? if so it should not be a warning but an error

Reply:
No, it's just a warning, it doesn't lead to a rule loading error. I reckon, there's some people using 'nocase' with this keyword, so it's best to just warn for now, I think at least :)